src/Moryx.Resources.UI.Interaction/Workspace/InteractionWorkspaceViewModel.cs

```diff
@@ -278,6 +278,10 @@ private async Task UpdateTreeAsync()
                 var refreshedTree = await treeTask;
                 await typeTask;
                 Tree.MergeTree(refreshedTree, new ResourceTreeMergeStrategy());
+
+                // EditMode currently blocks the Refresh button, so the second condition is only a precaution
+                if (SelectedResource != null && (ActiveItem as IResourceDetails)?.IsEditMode == false)
```

Comment:
Simply use `CurrentDetails.IsEditMode` instead of the cast

Reply:
Thanks for the hint. My thought was that UpdateTree is called from different events and I can't rely on `CanRefresh`. I know moved it to `Refresh`, since all other cases do not need to update the selected item.